lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -0,0 +1,147 @@
+ * and write to memory by SessionStore
+ * If EXISTING_PDU_SESSION, check on incoming session state and version
+ * accordingly take action.
+ * As per ASYNC method, response_callback is included but not functional
+ * */
+
+void SetMessageManagerHandler::SetAmfSessionContext(
+		ServerContext* context,
+		const SetSMSessionContext* request,
+		std::function<void(Status, SmContextVoid)> response_callback)
+{
+   auto& request_cpy = *request;
+   //Print the message from AMF
+   PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request_cpy));
+   response_callback(Status::OK, SmContextVoid());
+   /*The Event Based main_thread invocation and runs to handle session state*/ 
+   m5g_enforcer_->get_event_base().runInEventBaseThread (
+      [this, context, response_callback, request_cpy]() {
+   std::cerr << __LINE__ << " ACL_TAG entered event loop therad " << __FUNCTION__ << " processing message \n";
+   //extract values from proto
+   auto  imsi = request_cpy.common_context().sid().id();
+   std::string dnn = request_cpy.common_context().apn(); //may not required for demo-1
+   std::string session_ctx_id = id_gen_.gen_session_id(imsi);
+
+   MLOG(MDEBUG) << "Requested imsi from UE: " << imsi
+                 << " Generated sessioncontext ID" << session_ctx_id;
+   std::cerr << __LINE__ << " ACL_TAG Requested imsi from UE " << imsi << "\n";
+   /*reach complete message from proto message*/
+   SessionConfig cfg = m5g_build_session_config(request_cpy);
+   std::cerr << __LINE__ << " ACL_TAG m5g_build_session_config called \n";
+
+   /*check if it's initial message*/
+   if((cfg.rat_specific_context.m5gsm_session_context().rquest_type() == INITIAL_REQUEST) &&
+      (cfg.common_context.sm_session_state() == CREATING_0)) {
+      /*it's new UE establisment request and need to create the session context*/
+      	 MLOG(MDEBUG) << "AMF request type INITIAL_REQUEST and session state CREATING";
+      	 std::cerr << __LINE__ << " ACL_TAG AMF request type INITIAL_REQUEST and session state CREATING \n";
+         /* read the SessionMap from global session_store 
+	  * if not found it will add this imsi
+	  */
+	 SessionRead req = {imsi};
+         auto session_map = session_store_.read_sessions(req);
+	 send_create_session(session_map, imsi, session_ctx_id, cfg);
+    }
+  });
+}
+
+/* Creeate respective SessionState and context*/
+void SetMessageManagerHandler::send_create_session(
+		   SessionMap& session_map,
+		   const std::string& imsi,
+		   const std::string& session_ctx_id,
+		   const SessionConfig& cfg)
+{
+   auto session_map_ptr = std::make_shared<SessionMap>(std::move(session_map));
+   /* initialization of SessionState for IMSI by SessionStateEnforcer*/
+   bool success = m5g_enforcer_->m5g_init_session_credit(*session_map_ptr,
+		                                 imsi, session_ctx_id, cfg);
+   std::cerr << __LINE__ << " " << __FUNCTION__ << " successfully handled m5g_enforcer \n" ;

```

Comment:
Can we use MLOG instead here?

Reply:
All std::cerr will be removed from code once integration testing done. MLOG will be only way of logging traces.

Reply:
Also removed ACL_TAG from MLOG statements. It was added to filter traces.

Reply:
Next code commit, this std::cerr from code.